frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderView.tsx

```diff
@@ -30,69 +40,91 @@ import {
+type DashboardHeaderViewProps = {
   editingTitle?: string;
   editingSubtitle?: string;
   editingButtons?: JSX.Element[];
   editWarning?: string;
-  headerButtons?: ReactNode[];
-  headerClassName: string;
-  isEditing: boolean;
-  isEditingInfo: boolean;
-  isNavBarOpen: boolean;
   dashboard: Dashboard;
   collection: Collection;
   isBadgeVisible: boolean;
   isLastEditInfoVisible: boolean;
   onLastEditInfoClick: () => void;
-  setDashboardAttribute: <Key extends keyof Dashboard>(
-    key: Key,
-    value: Dashboard[Key],
-  ) => void;
-}
+} & DashboardFullscreenControls &
+  DashboardRefreshPeriodControls &
+  EmbedNightModeControls;
 
-export function DashboardHeaderComponent({
+export function DashboardHeaderView({
   editingTitle = "",
   editingSubtitle = "",
   editingButtons = [],
   editWarning,
-  headerButtons = [],
-  headerClassName = cx(CS.py1, CS.lgPy2, CS.xlPy3, CS.wrapper),
-  isEditing,
-  isNavBarOpen,
   dashboard,
   collection,
   isLastEditInfoVisible,
   onLastEditInfoClick,
-  setDashboardAttribute,
+  refreshPeriod,
+  onRefreshPeriodChange,
+  setRefreshElapsedHook,
+  isFullscreen,
+  onFullscreenChange,
+  isNightMode,
+  onNightModeChange,
+  hasNightModeToggle,
 }: DashboardHeaderViewProps) {
+  const isNavBarOpen = useSelector(getIsNavbarOpen);
+  const isEditing = useSelector(getIsEditing);
+
+  const handleSetDashboardAttribute = useSetDashboardAttributeHandler();
   const [showSubHeader, setShowSubHeader] = useState(true);
   const header = useRef<HTMLDivElement>(null);
   const dispatch = useDispatch();
 
   const isSidebarOpen = useSelector(getIsSidebarOpen);
   const isDashboardHeaderVisible = useSelector(getIsHeaderVisible);
+  const isAnalyticsDashboard = isInstanceAnalyticsCollection(collection);
 
   const _headerButtons = useMemo(
     () => (
       <HeaderButtonSection
         className="Header-buttonSection"
         isNavBarOpen={isNavBarOpen}
       >
-        {headerButtons}
+        <DashboardHeaderButtonRow
+          refreshPeriod={refreshPeriod}
+          onRefreshPeriodChange={onRefreshPeriodChange}
+          setRefreshElapsedHook={setRefreshElapsedHook}
+          isFullscreen={isFullscreen}
+          onFullscreenChange={onFullscreenChange}
+          isNightMode={isNightMode}
+          onNightModeChange={onNightModeChange}
+          hasNightModeToggle={hasNightModeToggle}
+          isAnalyticsDashboard={isAnalyticsDashboard}
+        />
       </HeaderButtonSection>
     ),
-    [headerButtons, isNavBarOpen],
+    [
+      hasNightModeToggle,
+      isAnalyticsDashboard,
+      isFullscreen,
+      isNavBarOpen,
+      isNightMode,
+      onFullscreenChange,
+      onNightModeChange,
+      onRefreshPeriodChange,
+      refreshPeriod,
+      setRefreshElapsedHook,
+    ],
   );
 
   const handleUpdateCaption = useCallback(
     async (name: string) => {
-      await setDashboardAttribute("name", name);
+      await handleSetDashboardAttribute("name", name);
```

Comment:
nit: I'd just keep the same name, I know this name is from moving the variable from another component into this file, but I don't see any problem with declaring

```ts
const setDashboardAttribute = useSetDashboardAttributeHandler()
```

since it's actually the same function.